ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerSubscription.scala

```diff
@@ -128,32 +123,38 @@ object ParallelIndexerSubscription {
       toMeteringDbDto: Iterable[(Offset, state.Update)] => Vector[DbDto.TransactionMetering],
   )(implicit
       loggingContext: LoggingContext
-  ): Iterable[((Offset, state.Update), Long)] => Batch[Vector[DbDto]] = { input =>
+  ): Iterable[(Offset, state.Update)] => Batch[Vector[DbDto]] = { input =>
+    val started = System.nanoTime()
     metrics.daml.parallelIndexer.inputMapping.batchSize.update(input.size)
-    input.foreach { case ((offset, update), _) =>
+    input.foreach { case (offset, update) =>
       withEnrichedLoggingContext("offset" -> offset, "update" -> update) {
         implicit loggingContext =>
           logger.info(s"Storing ${update.description}")
       }
     }
 
-    val mainBatch = input.iterator.flatMap { case ((offset, update), _) =>
+    val mainBatch = input.iterator.flatMap { case (offset, update) =>
       toDbDto(offset)(update)
     }.toVector
 
-    val meteringBatch = toMeteringDbDto(input.map(_._1))
+    val meteringBatch = toMeteringDbDto(input)
 
     val batch = mainBatch ++ meteringBatch
 
+    val nowNanos = System.nanoTime()
+    metrics.daml.parallelIndexer.inputMapping.duration.update(
+      (nowNanos - started) / input.size,
```

Comment:
here also I am not sure we need this particular metric at all. Can't we just use some EC/Thread-pool related metric already? @tudor-da

Reply:
(this applies to batching as well)

Reply:
not bad! I see that the `InstrumentedExecutorService` actually has a timer on `duration`. We should just check that the semantics are as precise as we need them, I don't see why they wouldn't be.

Reply:
yes it has!
what do you think about this @tudor-da `daml_parallel_indexer_inputmapping_executor_duration_mean{benchmark_id="$benchmark_id"}*irate(daml_parallel_indexer_inputmapping_executor_duration_count{benchmark_id="$benchmark_id"}[30s])`
@skisel-da can you think a way rationalizing the prometheus query, and the generated metrics? let's use something like this if possible! :+1: